user-service/src/main/java/rs/edu/raf/banka1/services/implementations/TransactionServiceImpl.java

```diff
@@ -60,14 +69,47 @@ public void createTransaction(Capital bankCapital, Capital securityCapital, Doub
             transaction.setSell(price);
             //Remove stocks
             capitalService.commitReserved(securityCapital.getListingId(), securityCapital.getListingType(), (double)securityAmount);
+            //checking how much tax we need to return
+            Double taxReturn = checkTaxReturn(order);
             //Add money
-            capitalService.addBalance(bankCapital.getCurrency().getCurrencyCode(), price);
+            capitalService.addBalance(bankCapital.getCurrency().getCurrencyCode(), price - taxReturn);
         }
         transaction.setMarketOrder(order);
         transaction.setEmployee(order.getOwner());
         transactionRepository.save(transaction);
     }
 
+    private Double checkTaxReturn(MarketOrder order){
+        List<MarketOrder> orders = orderRepository.getAllBuyOrders(order.getListingId(), order.getListingType(), order.getOwner(), OrderType.BUY, OrderStatus.DONE).orElse(null);
+        //this should not happen, checking just so java doesn't freak out
+        if(orders == null){
+            return 0.0;
+        }
+        Double returnAmount = 0.0;
+        Long counter = order.getContractSize();
+        for(MarketOrder buyOrder : orders){
+            long amount = Math.min(counter, buyOrder.getContractSize() - buyOrder.getCurrentAmount());
+            buyOrder.setCurrentAmount(amount - counter);
+            LocalDate dop = Instant.ofEpochSecond(buyOrder.getTimeStamp()).atZone(ZoneId.systemDefault()).toLocalDate();
```

Comment:
ako vec cuvamo timestamp, zasto ne bismo uzeli taj timestamp i currentTimestamp i videli koliko je vremena proslo

Reply:
to i radim?

Reply:
ali poredis localDate, mislim da cemo izbeci greske ako ostavimo na timestamp?

Reply:
vazi, promenicu na timestamp, oduzecu oba pa gledam da li je zbir veci ili manji od 10 godina u unix timestampu